spanner/client.go

```diff
@@ -424,11 +424,29 @@ func checkNestedTxn(ctx context.Context) error {
 func (c *Client) ReadWriteTransaction(ctx context.Context, f func(context.Context, *ReadWriteTransaction) error) (commitTimestamp time.Time, err error) {
 	ctx = trace.StartSpan(ctx, "cloud.google.com/go/spanner.ReadWriteTransaction")
 	defer func() { trace.EndSpan(ctx, err) }()
+	resp, err := c.rwTransaction(ctx, f, TransactionOptions{})
+	return resp.CommitTs, err
+}
+
+// ReadWriteTransactionWithOptions executes a read-write transaction with
+// configurable options, with retries as necessary.
+//
+// ReadWriteTransactionWithOptions is a configurable ReadWriteTransaction.
+//
+// See https://godoc.org/cloud.google.com/go/spanner#ReadWriteTransaction for
+// more details.
+func (c *Client) ReadWriteTransactionWithOptions(ctx context.Context, f func(context.Context, *ReadWriteTransaction) error, options TransactionOptions) (resp CommitResponse, err error) {
```

Comment:
Would it maybe be better to let the `TransactionOption` be an interface, and the `options` argument variadic? That would mean that:
1. We don't need a second `ReadWriteTransactionWithOptions` method, but can just change the `ReadWriteTransaction` method. This is still a breaking change, but only for some very specific use cases. See discussion in https://stackoverflow.com/questions/55163005/could-adding-variadic-parameters-to-a-function-break-existing-code.
1. Passing a `struct` means that you have to take into account what to do with the empty value of each field in the struct. See for example this article: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

This is also the approach taken by the `NewClient` function in [client.go](https://github.com/googleapis/google-cloud-go/blob/38a56325f4eed4a00a1661895369f69acf402d84/spanner/client.go#L136).

WDYT?

Reply:
@olavloite Yep, I have considered that as well. But the problem is the return type. We have to add a new method to return `CommitResponse`. So we can't just change the existing `ReadWriteTransaction` method.

Reply:
Ah, yes that is a good point.